### PR TITLE
FOGL-6799 Add Dryrun option to Python tasks

### PR DESCRIPTION
**Pull request by MarkRiddoch**
Signed-off-by: Mark Riddoch <mark@dianomic.com>

### CODE REVIEW

**Review by pintomax — APPROVED** [review]

**Review by pintomax — APPROVED** [review]

File: python/fledge/tasks/north/sending_process.py [hunk 1]
```diff
@@ -925,6 +925,9 @@ async def _start(self):
 
             exec_sending_process = self._config['enable']
 
+            if self.isDryRun():
+                return
```
Comment (ashish-jabble) [inline python/fledge/tasks/north/sending_process.py:929]: With --dryrun Sending process logs are coming with Started....Stop...Execution Completed, which means EGRESS of readings is going on if north server is reachable.

File: python/fledge/tasks/north/sending_process.py [hunk 2]
```diff
@@ -976,6 +975,11 @@ async def run(self):
         global _log_performance
         global _LOGGER
 
+        # Return early if task runs with --dryrun arg
+        # In future if any runtime configuration required for a task then it should be created before it
+        if self.is_dry_run():
```
Comment (praveen-garg) [inline python/fledge/tasks/north/sending_process.py:980]: move above globals?

**Review by praveen-garg — COMMENTED** [review]
> below logs I am getting on startup, not sure how to handle that. Still investigating!

We should be seeing these in develop too, NO? Scheduler on start calls _get_schedules (mind _ before it) and then schedule the first task and then next task... 

What is the concern here, to investigate?!

**Comment by ashish-jabble** [comment]
> We should be seeing these in develop too, NO?

Nope! 
With the given branch now on every startup you see the logs ONLY; not the actual task run as it early exit with dryrun mechanism.

**Comment by praveen-garg** [comment]
> not the actual task run as it early exit with dryrun mechanism.

You mean, now tasks are always running with dry run flag? I'm not following the earlier comment

**Comment by ashish-jabble** [comment]
> You mean, now tasks are always running with dry run flag? I'm not following the earlier comment

On every boot yes all of the tasks except STARTUP type called with dry-run flag. As asked in JIRA

**Review by ashish-jabble — APPROVED** [review]

**Pull request merged**